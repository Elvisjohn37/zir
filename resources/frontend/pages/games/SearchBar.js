import React, { useState } from 'react';
import Paper from '@mui/material/Paper';
import IconButton from '@mui/material/IconButton';
import InputBase from '@mui/material/InputBase';
import styles from './SearchBar.module';
import { eventDelay } from 'frontend/utils/helper';
import { injectIntl } from 'react-intl';
import { Search } from 'frontend/components/Icons';
import classnames from 'classnames';
import { isEmpty } from 'frontend/utils/helper';

let SearchBar = injectIntl(({ onChange, intl, toggable, value }) => {
	let searchInputRef = React.useRef();
	let [isActive, setIsActive] = useState(false);

	function onChangeWrapper() {
		onChange({ target: { value: searchInputRef.current.value } });
	}

	return (
		<>
			<Paper
				elevation={0}
				className={classnames([
					styles.searchPaper,
					toggable && styles.toggable,
					isActive || !isEmpty(value) ? styles.active : styles.inactive,
				])}
			>
				<div className={styles.searchInputContainer}>
					<InputBase
						inputRef={searchInputRef}
						placeholder={intl.messages['searchGame']}
						className={styles.searchInput}
						onKeyUp={() => {
							eventDelay(onChangeWrapper, 'gameSearch');
						}}
						onFocus={() => setIsActive(true)}
						onBlur={() => setIsActive(false)}
						size="small"
					/>
				</div>
				<div className={styles.searchButtonContainer}>
					<IconButton onClick={onChangeWrapper} className={styles.searchButton} size="small">
						<Search className={styles.searchIcon} />
					</IconButton>
				</div>
			</Paper>
			{toggable && (
				<Search
					className={classnames([styles.searchIconPlaceholder, isActive ? styles.active : styles.inactive])}
				/>
			)}
		</>
	);
});

export default SearchBar;
