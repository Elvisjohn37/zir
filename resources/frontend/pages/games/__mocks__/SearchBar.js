import { componentModuleMock } from 'frontend/tests/helpers';
module.exports = componentModuleMock(jest.createMockFromModule('../SearchBar'), 'SearchBar', {
	elementToUse: {
		default: 'input',
	},
});
