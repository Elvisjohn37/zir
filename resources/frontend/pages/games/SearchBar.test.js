import React from 'react';
import SearchBar from './SearchBar';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

jest.mock('@mui/material/Paper');
jest.mock('@mui/material/IconButton');
jest.mock('@mui/material/InputBase');
jest.mock('@mui/icons-material/Search');
jest.mock('frontend/utils/helper');
jest.mock('react-intl');

describe('<SearchBar/>', () => {
	it('should match snapshot', () => {
		let { asFragment } = render(<SearchBar onChange={jest.fn()} />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('should start search when user input any text', () => {
		let onChangeFn = jest.fn();
		let { getByTestId } = render(<SearchBar onChange={onChangeFn} />);
		let searchInput = getByTestId('InputBase');
		userEvent.type(searchInput, 'test');

		expect(onChangeFn).toHaveBeenCalledTimes(4);
		expect(onChangeFn).toHaveBeenNthCalledWith(1, { target: { value: 't' } });
		expect(onChangeFn).toHaveBeenNthCalledWith(2, { target: { value: 'te' } });
		expect(onChangeFn).toHaveBeenNthCalledWith(3, { target: { value: 'tes' } });
		expect(onChangeFn).toHaveBeenNthCalledWith(4, { target: { value: 'test' } });
	});

	it('should start search when user click search button ', () => {
		let onChangeFn = jest.fn();
		let { getByTestId } = render(<SearchBar onChange={onChangeFn} />);
		let searchButton = getByTestId('IconButton');
		userEvent.click(searchButton);

		expect(onChangeFn).toHaveBeenCalledTimes(1);
		expect(onChangeFn).toHaveBeenNthCalledWith(1, { target: { value: '' } });
	});
});
