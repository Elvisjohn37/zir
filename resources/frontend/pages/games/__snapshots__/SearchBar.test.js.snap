// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar/> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="searchPaper"
    data-testid="Paper"
    elevation="2"
  >
    <div
      class="searchInputContainer"
    >
      <input
        class="searchInput"
        data-testid="InputBase"
      />
    </div>
    <div
      class="searchButtonContainer"
    >
      <button
        class="searchButton"
        data-testid="IconButton"
      >
        <div
          data-testid="Search"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
